VimeoUpload/Upload/Networking/Shared/VimeoRequestSerializer+SharedUpload.swift

```diff
@@ -93,7 +94,48 @@ extension VimeoRequestSerializer
         
         return [Constants.SizeKey: fileSizeString]
     }
+    
+    private func createMIMETypeParameters(url: URL) throws -> [String: Any]
+    {
+        let type: String
+        do
+        {
+            type = try url.MIMEType()
+        }
+        catch let error as NSError
+        {
+            throw error.error(byAddingDomain: UploadErrorDomain.Create.rawValue)
+        }
+        return [Constants.MIMETypeKey: type]
+    }
 
+    func createFileParameters(url: URL) throws -> [String: Any]
+    {
+        let fileSizeParameters: [String: Any]
+        do
+        {
+            fileSizeParameters = try self.createFileSizeParameters(url: url)
+        }
+        catch let error as NSError
+        {
+            throw error.error(byAddingDomain: UploadErrorDomain.Create.rawValue)
+        }
+        
+        let fileMIMETypeParameters: [String: Any]
+        do
+        {
+            fileMIMETypeParameters = try self.createMIMETypeParameters(url: url)
+        }
+        catch let error as NSError
+        {
+            throw error.error(byAddingDomain: UploadErrorDomain.Create.rawValue)
+        }
+        
+        // Merge in the new key-value pairs passed in, favoring new values for any duplicate keys
+        let allFileParameters = fileSizeParameters.merging(fileMIMETypeParameters) { (current, new) in new }
+        return allFileParameters
```

Comment:
Similar comment as above. I would simply just return without assigning to a local constant.